src/plugins/intel_cpu/src/graph_optimizer.cpp

```diff
@@ -2424,196 +2582,233 @@ void GraphOptimizer::MergeTransposeAndReorder(Graph &graph) {
-
-    // Method checkAscendingSummaryOrder() checks that after the sequential execution of Transpose and Reorder nodes,
-    // the order of the elements in the memory will not change. In other words, that Transpose+Reorder is identical permutation.
-    auto checkAscendingSummaryOrder = [](std::shared_ptr<Node> &parentNode, std::shared_ptr<Node> &childNode) -> bool {
-        auto* transposeNode = dynamic_cast<Transpose*>(parentNode.get());
-        auto* reorderNode = dynamic_cast<Reorder*>(childNode.get());
-        if (!transposeNode || !reorderNode) {
+    auto isSuitableReshape = [](NodePtr node) {
+        if (node->getChildEdges().size() != 1)
             return false;
-        }
-
-        auto& transposeOrder = transposeNode->getOrder();
-        auto layoutOrder = transposeNode->getSelectedPrimitiveDescriptor()->getConfig().outConfs[0].getMemDesc()->as<BlockedMemoryDesc>()->getOrder();
-
-        auto inBlockedDesc = reorderNode->getSelectedPrimitiveDescriptor()->getConfig().inConfs[0].getMemDesc()->as<BlockedMemoryDesc>();
-        auto outBlockedDesc = reorderNode->getSelectedPrimitiveDescriptor()->getConfig().outConfs[0].getMemDesc()->as<BlockedMemoryDesc>();
-
-        auto& inOrder = inBlockedDesc->getOrder();
-        auto& outOrder = outBlockedDesc->getOrder();
-
-        if (transposeOrder.size() != layoutOrder.size() || layoutOrder.size() != inOrder.size() || inOrder.size() != outOrder.size()) {
+        // Reshape supported only in one case: if one of the input dims is split into 2 consecutive dims
+        const auto& inShape = node->getInputShapeAtPort(0).toPartialShape();
+        const auto& outShape = node->getOutputShapeAtPort(0).toPartialShape();
+        if (outShape.is_dynamic() || (outShape.size() - inShape.size() != 1))
             return false;
-        }
 
-        // revLayoutOrder - reverse permutation for layoutOrder
-        auto revLayoutOrder = VectorDims(layoutOrder.size());
-        for (size_t i = 0; i < revLayoutOrder.size(); i++) {
-            revLayoutOrder[layoutOrder[i]] = i;
+        size_t mismatchCount = 0;
+        for (size_t i = 0; i < inShape.size(); ++i) {
+            if (i + mismatchCount >= outShape.size())
+                return false;
+            if (inShape[i] != outShape[i + mismatchCount]) {
+                mismatchCount++;
+            }
         }
+        return mismatchCount == 1;
+    };
 
-        // newTransposeOrder - Transpose layout-aware permutation
-        auto newTransposeOrder = VectorDims(transposeOrder.size());
-        for (size_t i = 0; i < newTransposeOrder.size(); i++) {
-            newTransposeOrder[i] = layoutOrder[transposeOrder[revLayoutOrder[i]]];
-        }
+    auto isSuitableReorder = [](NodePtr node) {
+        return node->getType() == Type::Reorder
+                && node->getChildEdges().size() == 1
+                && !node->isDynamicNode();   // TODO [DS]: enable for dynamic shapes when inPlace in the dynamic case is available (CVS-74863)
```

Comment:
It looks we can fuse the Reorder and Transpose node with dynamic shapes now. We can remove the limitation if I understand correctly. @dmitry-gorokhov @maxnick Correct me.

Reply:
I thought twice. It looks not a simple job to enable the dynamic shape case, as Reorder uses a passthrough shapeinfer. But it is feasible when the Reorder node is optimized. One possible way is to keep reorder of how the reorder is transformed and apply it to shapeinfer.

Reply:
As I mentioned in [the comment](https://github.com/openvinotoolkit/openvino/pull/18999#discussion_r1389651862), we can support Reshape only in some specific cases. And to check that the Reshape satisfies this strict conditions, we need to know input/output shapes.
However, it is probably possible to support Reshape with one dynamic dimension in input/output shapes: in this case we still can understand whether the Reshape satisfies the conditions or not. What do you think?

Reply:
Yes, the dynamic shape restrictions cannot be easily relaxed. I won't insist on this. Let's consider it probably in a sperate story.